core/src/main/java/ai/timefold/solver/core/impl/score/stream/bavet/uni/BavetConcatUniConstraintStream.java

```diff
@@ -28,7 +28,7 @@ public BavetConcatUniConstraintStream(BavetConstraintFactory<Solution_> constrai
 
     @Override
     public boolean guaranteesDistinct() {
-        return false; // The two parents could have the same source; guarantee impossible.
```

Comment:
Unsure about `concat` class names for the same carnality; maybe do `BavetUniConcatUniConstraintStream` so it has the same format as all the others?